examples/gallery/database_module.py

```diff
@@ -0,0 +1,176 @@
+# change any attribute of a database entry. Here we edit the
+# wavemax attribute ( if it is less than 20.0, then we change it to 100.0;
+
+for database_entry in database:
+	if database_entry.wavemax < 20.0:
+		database.edit(database_entry,
+			wavemax=100.0)
+
+print(display_entries(database,
+	['id', 'observation_time_start', 'observation_time_end',
+	'instrument', 'wavemin', 'wavemax']))
+
+##############################################################################
+# Starring Entries. We can star(italic) database entries which are of interest
+# to us. Here we star all those entries with wavemin greater than 20.
+
+for database_entry in database:
+	if database_entry.wavemin > 20:
+		database.star(database_entry)
+
+print(display_entries(
+	filter(lambda entry: entry.starred, database),
+	['id', 'observation_time_start', 'observation_time_end',
+	'instrument', 'wavemin', 'wavemax']))
+
+##############################################################################
+# Adding tags to entries. By tagging entries we can add more information to
+# each entry in the form of keywords. The tags(bold) property of a database
+# object holds all the tags saved in the database. Here we tag all entries
+# whose observation_time_start was in year 2011.
+
+for database_entry in database:
+	if database_entry.observation_time_start.year == 2011:
+		database.tag(database_entry, 'eleven')
+
+print(database.tags)
+
+sample_tag = database.tags[0]
+print(display_entries(
+	filter(lambda entry: sample_tag in entry.tags, database),
+	['id', 'observation_time_start', 'observation_time_end',
+	'instrument', 'wavemin', 'wavemax']))
+
+##############################################################################
+# Undo and redo operations. These features are very useful as they can undo or
+# redo the last n commands. Note : If we work on a Database object in an 
+# interactive Python session and quit this session, the undo and redo history
+# are lost. Here we undo and redo the last 2 commands.
+
+database.undo(2)
+database.redo(2)
+
+##############################################################################
+# Removing entries. We use the database.remove() method to remove a database
+# entry. Here we remmove all those entries whose observation_time_start is
+# in year 2012.
+
+for database_entry in database:
+	if database_entry.observation_time_start.year == 2012:
+		database.remove(database_entry)
+print(display_entries(database,
+	['id', 'observation_time_start', 'observation_time_end',
+	'instrument', 'wavemin', 'wavemax']))
+
+##############################################################################
+# Emptying the database. The clear() method can be used to remove all the
+# entries in the database.
+
+database.clear()
+print(len(database))
+
+##############################################################################
+# Committing changes to the database. While working in an interactive python
```

Comment:
Commit is only required when you want to save the changes made to the database. Only after committing, the database file on the disk will be updated.